File: summer/WebContent/js/system/componentmodel/IItemProperties.js
/**
 * IItemProperties
 */

define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	var IItemProperties = declare("IItemProperties", null,{
	});
	
	Object.defineProperties(IItemProperties.prototype,{
//		ReadOnlyCollection<ItemPropertyInfo> 
		ItemProperties:
		{
			get:function(){}
		}
	});
	
	IItemProperties.Type = new Type("IItemProperties", IItemProperties, [Object.Type]);
	return IItemProperties;
});

File: org.summer.view.widget/src/org/summer/view/widget/model/IItemProperties.java
package org.summer.view.widget.model;

import org.summer.view.widget.collection.ReadOnlyCollection;

/// <summary>
/// IItemProperties is an interface that a collection view
/// can implement to expose information about the properties available on
/// items in the underlying collection.
/// </summary>
public interface IItemProperties
{
    /// <summary>
    /// Returns information about the properties available on items in the
    /// underlying collection.  This information may come from a schema, from
    /// a type descriptor, from a representative item, or from some other source
    /// known to the view.
    /// </summary>
    ReadOnlyCollection<ItemPropertyInfo>    ItemProperties { get; }
}
  
/// <summary>
/// Information about a property.  Returned by <seealso cref="IItemProperties.ItemProperties">
/// </seealso></summary>
